core/coreutils.cpp

```diff
@@ -214,7 +214,12 @@ QString CoreUtils::generateConflictedCopyFileName( const QString &file, const QS
 
   QFileInfo f( file );
 
-  return QString( "%1/%2 (conflicted copy, %3 v%4).%5" ).arg( f.path(), f.baseName(), username, QString::number( version ), f.completeSuffix() );
+  QString suffix = f.completeSuffix();
+  if ( !suffix.compare( QStringLiteral( "qgs" ), Qt::CaseInsensitive ) || !suffix.compare( QStringLiteral( "qgz" ), Qt::CaseInsensitive ) )
```

Comment:
We have a function in Mergin API called `hasProjecFileExtension`, can you please move it here to coreutils, fix typo in its name and use it instead of checking suffixes here again? 🙂  Thanks! 

It would also be good to have a test for that function.

Reply:
Done